content/docs/infrastructure-wg.md

```diff
@@ -0,0 +1,196 @@
+  who are not paid for their contirbutions by some company or organisation.
+* Contractor: In case something is very large or we can't find anyone to do it
+  the work can be handed off to a contractor.
+
+
+## Ideas for initial work items
+
+This is an unsorted, unprioritized list of ideas the working group could look
+into: 
+
+* Run a survey in the community to understand the largest pain points and 
+  prioritze potential improvements.
+* Create and maintain a public documentation of the existing infrastructure and 
+  the supported workflows (what, why, who), including down-stream usages of 
+  LLVM.
+* Define and agree a vision for LLVM infrastructure and for the design 
+  principles thereof: What should the development workflow and the 
+  infrastructure look like?
+* Identify the need for addons for Github the community is missing right now 
+  (e.g. subscribing to labels on Github Issues).
+* Come to a decision on using Phabricator vs. GitHub Pull Requests for code
+  reviews, then implement the change.
+* Create a roadmap for the handover of the existing infrastructure to a 
+  contractor. Components ready for immediate takeover:
+  * Phabricator (in case we stay with it)
+  * Pre-merge testing
+  * Buildbot workers
+  * clangd remote index for LLVM
+  * [LNT service](http://lnt.llvm.org/)
+  * ...
+* Create new design and structure for the LLVM website, migrate the existing 
+  content there.
+* Create a test strategy for the LLVM project and refactor the existing build 
+  infrastructure to meet the new test strategy. Some ideas:
+  * Multi-stage CI: run “cheap” builds first, run “expensive” builds only on
+    commits where the cheap builds have passed.
+  * Set up a collector dashboard showing build results for all CI systems per 
+    commit, e.g. Treeherder or something GitHub based.
+  * Investigate if we can get a good deal from GitHub Actions with sponsored 
+    hardware.
+  * Create a plan of the "important" buildbot configuration we need in order to 
+    detect the bugs with large impact (contributors and downstream). Then have
+    one central team/contractor operate these to ensure consistency and quality.
+* Gather user feedback on the existing infrastructure: Where are the pain 
+  points? What would our community like to see improved?
+* Migration of all python scripts and tools to Python 3 as Python 2 is 
+  deprecated.
+* Reach out to downstream infrastructure teams (e.g. Rust, Tensorflow) and 
+  understand if we have infrastructure topics we want to cooperate on.
+* Define Service Level Objectives (SLO) for our infrastructure and monitor 
+  them. Set up monitoring and altering for our infrastructure.  Derive measure
+  where we do not meet our SLOs.
+* Migrate Windows setup instructions off GnuWin32 as it’s deprecated and add
+  installation instructions users can copy-and-paste into their shell.
+* Investigate the Rust project infrastructure and see what would be helpful to 
```

Comment:
Good idea. The Rust team has some kind of deal with the github folks on github actions: https://blog.rust-lang.org/inside-rust/2020/07/23/rust-ci-is-moving-to-github-actions.html